ml_ms4alg/ms4alg.py

```diff
@@ -685,27 +693,27 @@ def sort(self):
                 os.remove(fname0)
             NS.setHdf5FilePath(fname0)
             neighborhood_sorters.append(NS)
+            tmp_dask = dask.delayed(run_phase1_sort(NS))
+            dask_list.append(tmp_dask)
+
+        dask.compute(*dask_list, num_workers=self._num_workers)
 
-        pool = multiprocessing.Pool(num_workers)
-        pool.map(run_phase1_sort, neighborhood_sorters)
-        pool.close()
-        pool.join()
         #for m in range(M):
         #    print ('Running phase1 neighborhood sort for channel {} of {}...'.format(m+1,M)); sys.stdout.flush()
         #    neighborhood_sorters[m].runPhase1Sort()
-
+        dask_list = []
         for m in range(M):
             times_m=neighborhood_sorters[m].getPhase1Times()
             channel_assignments_m=neighborhood_sorters[m].getPhase1ChannelAssignments()
             for m2 in range(M):
                 inds_m_m2=np.where(channel_assignments_m==m2)[0]
                 if len(inds_m_m2)>0:
                     neighborhood_sorters[m2].addAssignedEventTimes(times_m[inds_m_m2])
+        for m in range(M):
+            tmp_dask = dask.delayed(run_phase2_sort(neighborhood_sorters[m]))
```

Comment:
```suggestion
        for m in neighborhood_sorters[:M]:
            tmp_dask = dask.delayed(run_phase2_sort(sorter))
```

Reply:
Or replace `neightborhood_sorters[:M]` with `itertools.islice(neighborhood, M)` if you expect M to be large and care about performance.